File: AkkaSingleFs/SimpleSetup.fs
module SimpleSetup

open System
open Akka.Actor

type ConsoleWriterActor() =
    inherit UntypedActor()
    
    override my.OnReceive message =
        printfn $"Retrieved: %A{message}"
        
type Runner(writer: IActorRef) =
    inherit UntypedActor()
    
    override my.OnReceive _ =
        let text = Console.ReadLine()
        if String.IsNullOrEmpty text then
            // Unlike C#, static member must be explicit with type.
            Runner.Context.System.Terminate() |> ignore
        else
            writer.Tell text
            my.Self.Tell "continue"

let Example (actor_system: ActorSystem) =
    let write_actor = actor_system.ActorOf(Props.Create<ConsoleWriterActor>())
    let runner = actor_system.ActorOf(Props.Create(fun _ -> Runner write_actor))
    
    printfn "Input texts. Just ENTER to end."
    runner.Tell "start"

File: AkkaSingle/SimpleSetup.cs
using Akka.Actor;

namespace AkkaSingle
{
    sealed class ConsoleWriterActor : UntypedActor
    {
        protected override void OnReceive(object message) {
            Console.WriteLine("Retrieved: {0}", message);
        }
    }

    sealed class Runner : UntypedActor
    {
        readonly IActorRef writer;
        public Runner(IActorRef writer) {
            this.writer = writer;
        }

        protected override void OnReceive(object message) {
            var text = Console.ReadLine();
            if (string.IsNullOrEmpty(text)) {
                Context.System.Terminate();
                return;
            }

            writer.Tell(text);

            Self.Tell("continue");
        }
    }

    // Basically, this is a short version of Akka Bootcamp unit-1
    // ref: https://github.com/petabridge/akka-bootcamp/blob/master/src/Unit-1/lesson1/README.md
    static class SimpleSetup
    {
        public static void Example(ActorSystem actorSystem) {
            var writeActor = actorSystem.ActorOf(Props.Create(() => new ConsoleWriterActor()));
            var runner = actorSystem.ActorOf(Props.Create(() => new Runner(writeActor)));

            Console.WriteLine("Input texts. Just ENTER to end.");
            runner.Tell("start");
        }
    }
}